src/FlatSharp.Runtime/IO/MemoryInputBuffer.cs

```diff
@@ -25,12 +25,14 @@ public struct MemoryInputBuffer : IInputBuffer
 {
     private readonly MemoryPointer pointer;
 
-    public MemoryInputBuffer(Memory<byte> memory)
+    public MemoryInputBuffer(Memory<byte> memory, bool pinned = false)
```

Comment:
@jamescourtney I added `IInputBuffer.IsPinned` and added parameter with default value to constructor of `MemoryInputBuffer` and `ReadOnlyMemoryInputBuffer` as I that was practical approach. I am fine adding a new buffer type and/or removing the `IsPinned` and keying off the type. Let me know what you think.

Reply:
I think IsPinned is a great compromise.